zebra-chain/src/primitives/address.rs

```diff
@@ -185,10 +163,34 @@ impl Address {
             Self::Transparent(address) => Some(address.to_string()),
             Self::Sapling { address, network } => {
                 let data = address.to_bytes();
-                let address = ZcashAddress::from_sapling(network.into(), data);
+                let address = ZcashAddress::from_sapling(network.to_zcash_address(), data);
                 Some(address.encode())
             }
             Self::Unified { .. } => None,
         }
     }
 }
+
+impl NetworkKind {
+    /// Converts a [`zcash_address::Network`] to a [`NetworkKind`].
+    ///
+    /// This method is meant to be used for decoding Zcash addresses in zebra-rpc methods.
+    fn from_zcash_address(network: zcash_address::Network) -> Self {
+        match network {
+            zcash_address::Network::Main => NetworkKind::Mainnet,
+            zcash_address::Network::Test => NetworkKind::Testnet,
+            zcash_address::Network::Regtest => NetworkKind::Regtest,
+        }
+    }
+
+    /// Converts a [`zcash_address::Network`] to a [`NetworkKind`].
+    ///
+    /// This method is meant to be used for encoding Zcash addresses in zebra-rpc methods.
+    fn to_zcash_address(self) -> zcash_address::Network {
+        match self {
+            NetworkKind::Mainnet => zcash_address::Network::Main,
+            NetworkKind::Testnet => zcash_address::Network::Test,
```

Comment:
This may still need another `NetworkKind` variant (which includes a String for the network part of the address prefix), or it may be fine as it is, let's ask if it's acceptable to use the `ztest` part of the prefix for configurable testnet addresses.

Reply:
It's okay to use the Testnet HRPs for now, we should add configurable HRPs once https://github.com/zcash/librustzcash/issues/1344 is implemented.